operator/v1/types_ingress.go

```diff
@@ -513,6 +520,59 @@ type IngressControllerLogging struct {
 	Access *AccessLogging `json:"access,omitempty"`
 }
 
+// IngressControllerHTTPHeaderPolicy is a policy for setting HTTP headers.
+//
+// +kubebuilder:validation:Enum=Append;Replace;IfNone;Never
+type IngressControllerHTTPHeaderPolicy string
+
+const (
+	// AppendHTTPHeaderPolicy appends the header, preserving any existing header.
+	AppendHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "Append"
+	// ReplaceHTTPHeaderPolicy sets the header, removing any existing header.
+	ReplaceHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "Replace"
+	// IfNoneHTTPHeaderPolicy sets the header if it is not already set.
+	IfNoneHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "IfNone"
+	// NeverHTTPHeaderPolicy never sets the header, preserving any existing
+	// header.
+	NeverHTTPHeaderPolicy IngressControllerHTTPHeaderPolicy = "Never"
+)
+
+// IngressControllerHTTPForwardedHeaderPolicy describes how the HTTP Forwarded
+// header and related headers are handled.
+type IngressControllerHTTPForwardedHeaderPolicy struct {
+	// policy specifies when and how the IngressController sets the
+	// Forwarded, X-Forwarded-For, X-Forwarded-Host, X-Forwarded-Port,
+	// X-Forwarded-Proto, and X-Forwarded-Proto-Version headers.  The value
+	// may be one of the following:
+	//
+	// * "Append", which specifies that the IngressController appends the
+	//   headers, preserving existing headers.
+	//
+	// * "Replace", which specifies that the IngressController sets the
+	//   headers, replacing any existing Forwarded or X-Forwarded-* headers.
+	//
+	// * "IfNone", which specifies that the IngressController sets the
+	//   headers if they are not already set.
+	//
+	// * "Never", which specifies that the IngressController never sets the
+	//   headers, preserving any existing headers.
+	//
+	// By default, the policy is "Append".
+	//
+	// +optional
+	Policy IngressControllerHTTPHeaderPolicy `json:"policy,omitempty"`
+}
+
+// IngressControllerHTTPHeaders specifies how the IngressController handles
+// certain HTTP headers.
+type IngressControllerHTTPHeaders struct {
```

Comment:
What else would theoretically go into this struct as compared to what would theoretically get added to `IngressControllerHTTPForwardedHeaderPolicy`.  Are these truly distinct structs?  Maybe the next two most likely fields for each struct would help me see it.

Reply:
The next field we'd add is `uniqueId`, for injecting a custom, unique header into every request for tracing:  https://github.com/openshift/api/pull/689/commits/43ad0aa38a5036646ce42ab1c5fb16afebea80e4#diff-85d84174432a6d2df99d107129e012f8R601-R611

Beyond that, we have an RFE to allow setting arbitrary headers, such as `Strict-Transport-Security` and `X-XSS-Protection`, with static values.  To implement that, I expect we'll add a field with a slice type for specifying arbitrary name-value pairs.

Reply:
miciah also came up with "Maybe an OmitNonStandard (the X-Forwarded-*" for the ForwardedHeaderPolicy.

@Miciah if nothing gets added to this in the next year, we should remember that in the future.